src/engine/controls/cuecontrol.cpp

```diff
@@ -2168,11 +2174,106 @@ void CueControl::updateIndicators() {
             m_pCueIndicator->setBlinkValue(ControlIndicator::ON);
         }
     }
+
+    // Update hotcue indicators: activate the indicator if we are at the cue pos
+    // or if we passed it since the last update, else deactivate.
+    auto prevPos = m_prevPosition.getValue();
+    const auto currPos = frameInfo().currentPosition;
+    VERIFY_OR_DEBUG_ASSERT(prevPos.isValid()) {
+        m_prevPosition.setValue(currPos);
+        return;
+    }
+
+    // If we're looping check if we just wrapped around.
+    // Since explicit seeks are reported in notifiySeek(), this is safe
+    // to detect the true loop wrap-around.
+    LoopInfo loopInfo = m_loopInfo;
+    bool loopingWrapAround = false;
+    bool loopingWrapAroundReverse = false;
+    bool reverse = m_pReverse->toBool();
+    if (m_pLoopEnabled->toBool() &&
+            loopInfo.startPosition.isValid() &&
+            loopInfo.endPosition.isValid() &&
+            // Are we inside the current loop?
+            prevPos >= loopInfo.startPosition &&
+            currPos >= loopInfo.startPosition &&
+            prevPos < loopInfo.endPosition &&
+            currPos < loopInfo.endPosition) {
+        // Check if we wrapped around
+        if (reverse && prevPos < currPos) {
```

Comment:
Doesn't the condition apply to any seek, independent of a loop? 

I am just considering if it will work more universal we just check if cue position is inside the range of prevPos + the buffer length? 

This will not work if the cue is just after the current loop though. 
In case of seeks the buffer before the seek is cossfaded with the buffer after the seek. Is it inside a loop? 
What is the use case for such a corner case?

Reply:
> Doesn't the condition apply to any seek, independent of a loop?

Seeks are registered in notifySeek() which sets prevPos(), i.e. the loop detection doesn't kick in.

> check if cue position is inside the range of prevPos + the buffer length?

Since the buffer is split into two parts if we wrap around, that's exactly what we do here. (besides, I also considered using LoopingControl's `nextTrigger()` like ReadAheadManager does, or get the trigger already in EngineBuffer so it can pass it to CueControl and RAManager, but that is overkill IMO)

Actually, the original implementation worked just fine (incl. scratching and reverse, NOT for loops),
the current implementation works also for loops, but fails for scratching backwards inside a loop.

Back to square one...

Reply:
okay, "speed" from EngineBuffer works: it is "effective" reverse, i.e. `reverse` or scratching backwards.
now everything works as expected.